src/components/AnalysisModal.js

```diff
@@ -117,6 +117,10 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
         resetView()
         onDismiss()
       },
+      onError: () => {
```

Comment:
may be worth defining onDismiss above in this case as to not duplicate the code

Reply:
How does the modalDismiss function look to you?